source/common/upstream/ring_hash_lb.cc

```diff
@@ -116,8 +120,13 @@ RingHashLoadBalancer::Ring::Ring(
 
       // Sadly std::hash provides no mechanism for hashing arbitrary bytes so we must copy here.
       // xxHash is done wihout copies.
-      const uint64_t hash = use_std_hash ? std::hash<std::string>()(std::string(hash_key))
-                                         : HashUtil::xxHash64(hash_key);
+      const uint64_t hash =
+          use_std_hash
+              ? std::hash<std::string>()(std::string(hash_key))
+              : (hash_function == HashFunction::Cluster_RingHashLbConfig_HashFunction_MURMUR_HASH_2)
+                    ? HashUtil::murmurHash2_64(hash_key, HashUtil::STD_HASH_SEED)
```

Comment:
`HashUtil::murmurHash2_64(hash_key, HashUtil::STD_HASH_SEED)` is a drop-in replacement for `std::hash<std::string>()(std::string(hash_key))`, so why are we keeping `std::hash` code path? Shouldn't you replace code path for `use_std_hash` with `murmurHash2_64`?

Reply:
We aren't, this is getting deleted and is already deprecated. The idea here is to give everyone using GCC a stable way to move forward without having to rehash (the fact that we ever used std::hash here was my horrible mistake).

Reply:
What I'm saying is that because `murmurHash2` is a drop-in replacement, we can already remove `std::hash` from the code, while still leaving the deprecated `use_std_hash` working at the API level, i.e.:
```
const uint64_t hash =
    (use_std_hash ||
     hash_function == HashFunction::Cluster_RingHashLbConfig_HashFunction_MURMUR_HASH_2)
        ? HashUtil::murmurHash2_64(hash_key, HashUtil::STD_HASH_SEED)
        : HashUtil::xxHash64(hash_key);
```

Reply:
The reason I suggest doing it this way is that it isn't guaranteed that `std::hash` is murmur2. It happens to be that way on at least GCC5 - GCC7 stdlib, but it isn't that way on OSX clang for example hence the ifdefs in the code. IMO we should do it this way so that we don't change `std::hash` accidentally. The great majority of people that are using GCC stdlib will have an out when we deprecate.

Reply:
Ah, OK. Makes sense now, thanks!